src/bejeweled/board/Board.java

```diff
@@ -256,41 +258,90 @@ private boolean checkForCombinations(Gem gem) {
 		}
 		return false;
 	}
+	
+	public ArrayList<Combination> toCombinations(ArrayList<Gem> g) {
```

Comment:
It is really hard to implement a method that can split for example the red ArrayList into multiple combinations. I also think it will take too much time when running the game because you need a lot of loops, also inside eachother. That's why I decided not to continue implement it, however I have tried it. And in my experience so far it's not needed at all.